tests/test_params_ecmwf.F90

```diff
@@ -15,6 +15,10 @@ program test_params_ecmwf
   if (g2disc .ne. 255 .or. g2cat .ne. 255 .or. g2num .ne. 255) stop 2
   call param_ecmwf_g1_to_g2(52, 1, g2disc, g2cat, g2num)
   if (g2disc .ne. 0 .or. g2cat .ne. 1 .or. g2num .ne. 1) stop 3
+  call param_ecmwf_g1_to_g2(-52, -1, g2disc, g2cat, g2num)
+  if (g1val .le. 0 .or. g1ver .le. 0) stop 5
```

Comment:
You need to check the values of g2disc, g2cat, g2num after each call to param_ecmwf_g1_to_g2().